### PR TITLE
Modifications to filters structure

### PR DESCRIPTION
**Pull request by CremaLuca**
## Premise
As discussed in this [Trello card](https://trello.com/c/ajkA5sbh) I've implemented the required changes to Filters and FilterList (that will probably be renamed into FilterGraph soon)

#### feat(`streams` as `strings`)
When defining a new instance of a filter stream instances are no longer required as parameters, instead, now stream names are used and streams are given to the filter only later.
As already discussed this has some ups and downs:
**+** it's way easier to write and concatenate streams when defining a `FilterList` instance.
**~** requires some more code inside the `Filter`.
These modifications were mainly applied to `FilterList` and all of the Filters.

#### feat(`setup` method)
Filters must receive the streams they should be working on at some point, it was either inside of the `execute()` method or create a `setup()`, the latter has been chosen.
This method must be called before the first call to the `execute()` and lets the filter save instances of streams and their iterators. Furthermore, a dictionary called "status" is being passed, I'll talk about it later.

#### feat(`state` as parameter)
It's been agreed that no instance of filters should be saved by the writer of `FilterList`s, so the state must be saved inside `FilterList` in order to be able to access it. That's been implemented by keeping a dictionary that is passed to filters in the `setup` method that the filter will use to save its variables it might want to be outputted.

#### test
Oh my God, you gotta see how many tests I had to change. Please spare me if I removed some (quite useless, things that were already tested in `Filter`)

## Picture of a cute animal
You may rest now before you pass to the 23+ changed files.
Cute

### CODE REVIEW

File: otri/filtering/filters/phase_filter.py [hunk 1]
```diff
@@ -27,15 +31,32 @@ def __init__(self, input_stream: Stream, keys_to_change: Mapping[str, Callable[[
...
+            output=[output],
+            input_count=1,
+            output_count=1
+        )
         self.__keys = keys_to_change
         self.__distance = distance
         self.__atoms_buffer = list()
         self.__counter = 0
 
+    def setup(self, inputs: Sequence[Stream], outputs: Sequence[Stream], status: Mapping[str, Any]):
```
Comment (CremaLuca) [inline otri/filtering/filters/phase_filter.py:45]: Codacy Issue found: [Signature differs from overridden 'setup' method](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 21
           

Complexity increasing per file
==============================
- test/filtering/filters/split_filter_test.py  1
         

Complexity decreasing per file
==============================
+ test/filtering/filter_test.py  -1
+ otri/filtering/filter_list.py  -7
+ otri/filtering/filters/statistics_filter.py  -1
+ autocorrelation.py  -2
         

Clones added
============
- test/filtering/filters/merge_filter_test.py  3
- otri/filtering/filters/math_filter.py  2
- otri/filtering/filters/nuplicator_filter.py  1
- otri/filtering/filters/statistics_filter.py  2
- otri/filtering/filters/generic_filter.py  2
- otri/filtering/filters/interpolation_filter.py  2
- otri/filtering/filters/phase_filter.py  2
- test/filtering/filters/split_filter_test.py  2
         

Clones removed
==============
+ otri/filtering/filters/split_filter.py  -1
+ test/filtering/filters/statistics_filter_test.py  -2
+ autocorrelation.py  -2
+ test/filtering/filters/phase_filter_test.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486&bid=18499171)

File: otri/filtering/filters/generic_filter.py [hunk 2]
```diff
@@ -1,42 +1,61 @@
...
-            output_streams_count=1
+            input=[input],
+            output=[output],
+            input_count=1,
+            output_count=1
         )
         self.__operation = operation
-        self.__source_iter = source_stream.__iter__()
+
+    def setup(self, inputs : Sequence[Stream], outputs : Sequence[Stream], status: Mapping[str, Any]):
```
Comment (CremaLuca) [inline otri/filtering/filters/generic_filter.py:33]: Codacy Issue found: [Signature differs from overridden 'setup' method](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/statistics_filter.py [hunk 3]
```diff
@@ -1,167 +1,178 @@
...
+    and outputs the data unmodified.
 
-    def __init__(self, input_stream: Stream, keys: Collection[str]):
+    Input:
+        Single stream.
+    Output:
+        Single stream.
+    '''
+
+    def __init__(self, input: str, output: str, keys: Collection[str]):
```
Comment (CremaLuca) [inline otri/filtering/filters/statistics_filter.py:17]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/math_filter.py [hunk 4]
```diff
@@ -1,42 +1,60 @@
...
-        self.__input_stream_iter = input_stream.__iter__()
-        self.__output_stream = self.get_output_stream(0)
+        super().__init__(
+            input=[input],
+            output=[output],
+            input_count=1,
+            output_count=1)
         self.__keys_operations = keys_operations
 
+    def setup(self, inputs: Sequence[Stream], outputs: Sequence[Stream], status: Mapping[str, Any]):
```
Comment (CremaLuca) [inline otri/filtering/filters/math_filter.py:31]: Codacy Issue found: [Signature differs from overridden 'setup' method](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/generic_filter.py [hunk 5]
```diff
@@ -1,42 +1,61 @@
...
+    Applies a given Callable to all input data and outputs them in the output stream.
+    
+    Input:
+        Single stream.
+    Outputs:
+        Single stream.
     '''
 
-    def __init__(self, source_stream: Stream, operation: Callable):
+    def __init__(self, input: str, output: str, operation: Callable):
```
Comment (CremaLuca) [inline otri/filtering/filters/generic_filter.py:15]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/phase_filter.py [hunk 6]
```diff
@@ -1,24 +1,28 @@
...
-        A single Stream containing n - c atoms, whose fields are the
+        Single stream containing n - c atoms, whose fields are the
         result of the given operation between the atoms at positions
-        i and i+c in the input Stream. Such fields are the only fields.
+        i and i+c in the input Stream. Such fields are the only fields
+        of the output data.
     '''
 
-    def __init__(self, input_stream: Stream, keys_to_change: Mapping[str, Callable[[Any, Any], Any]], distance: int):
+    def __init__(self, input: str, output: str, keys_to_change: Mapping[str, Callable[[Any, Any], Any]], distance: int):
```
Comment (CremaLuca) [inline otri/filtering/filters/phase_filter.py:18]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/nuplicator_filter.py [hunk 7]
```diff
@@ -1,31 +1,49 @@
...
-            output_streams_count=output_streams_count
+            input=[input],
+            output=output,
+            input_count=1,
+            output_count=len(output)
         )
         self.__copy = copy.deepcopy if deep_copy else copy.copy
-        self.__source_iter = source_stream.__iter__()
+
+    def setup(self, inputs : Sequence[Stream], outputs : Sequence[Stream], status: Mapping[str, Any]):
```
Comment (CremaLuca) [inline otri/filtering/filters/nuplicator_filter.py:33]: Codacy Issue found: [Signature differs from overridden 'setup' method](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/split_filter.py [hunk 8]
```diff
@@ -16,38 +15,70 @@ class SplitFilter(Filter):
     will be the last one (of index n+1).
     '''
 
-    def __init__(self, source_stream: Stream, key: Any, ranges: Sequence, ignore_none: bool = True, side: str = 'left'):
+    def __init__(self, input: str, output : Sequence[str], key: Any, ranges: Sequence, none_keys_output: str = None, side: str = 'left'):
```
Comment (CremaLuca) [inline otri/filtering/filters/split_filter.py:18]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/statistics_filter.py [hunk 9]
```diff
@@ -1,167 +1,178 @@
...
+            input_count=1,
+            output_count=1
         )
-        # Dict like : {callable : dict}
         self.__keys = keys
+        # Dict like : {callable : status_name}
         self.__ops = dict()
-        self.__input_iter = input_stream.__iter__()
+
+    def setup(self, inputs : Sequence[Stream], outputs : Sequence[Stream], status: Mapping[str, Any]):
```
Comment (CremaLuca) [inline otri/filtering/filters/statistics_filter.py:37]: Codacy Issue found: [Signature differs from overridden 'setup' method](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/split_filter.py [hunk 10]
```diff
@@ -56,25 +87,23 @@ def execute(self):
...
                 # Ignoring the item that does not have the key.
-                if self.__ignore_none:
-                    self.execute()
-                # Putting the item in the dedicated Stream.
-                else:
-                    self.__none_stream.append(item)
-        elif self.get_input_stream(0).is_closed():
+                if not self.__ignore_none:
+                    # Append void atom on last output
+                   self.__none_output.append(item)
```
Comment (CremaLuca) [inline otri/filtering/filters/split_filter.py:103]: Codacy Issue found: [Bad indentation. Found 19 spaces, expected 20](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/merge_filter.py [hunk 11]
```diff
@@ -1,41 +1,59 @@
...
-        # caching iterators to avoid creating a new one every time execute() is called
-        self.input_iterators = [x.__iter__() for x in input_streams]
-        self.output_stream = self.get_output_stream(0)
+        super().__init__(
+            input=input,
+            output=[output],
+            input_count=len(input),
+            output_count=1)
+
+    def setup(self, inputs: Sequence[Stream], outputs: Sequence[Stream], status: Mapping[str, Any]):
```
Comment (CremaLuca) [inline otri/filtering/filters/merge_filter.py:28]: Codacy Issue found: [Signature differs from overridden 'setup' method](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filter.py [hunk 12]
```diff
@@ -7,73 +7,73 @@ class Filter:
...
-            Streams where manipulated atoms are placed.
+        input : Sequence[str]
+            Name for input streams. If there are multiple streams the filter must define the right order.
+            Streams will be gathered inside the FilterList's dictionary of streams.
+        output : Sequence[str]
+            Name for output streams,
     '''
 
-    def __init__(self, input_streams: Collection[Stream], input_streams_count: int = 0, output_streams_count: int = 0):
+    def __init__(self, input: Sequence[str], output: Sequence[str], input_count: int = 0, output_count: int = 0):
```
Comment (CremaLuca) [inline otri/filtering/filter.py:17]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/split_filter.py [hunk 13]
```diff
@@ -89,47 +118,74 @@ class SwitchFilter(Filter):
     ordering is not.
     '''
 
-    def __init__(self, source_stream: Stream, key: Any, cases: Set, ignore_none: bool = True):
+    def __init__(self, input: str, cases_output : Sequence[str], default_output : str, key: Any, cases: Set, none_keys_output: str = None):
```
Comment (CremaLuca) [inline otri/filtering/filters/split_filter.py:121]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filter_list.py [hunk 14]
```diff
@@ -1,90 +1,92 @@
...
-        while(not self.__is_all_finished()):
-            for filter_layer in self.layers:
-                for fil in filter_layer:
-                    fil.execute()
-            # If any filter has outputted any atom, call the on_atom_output method
-            if on_atom_output != None:
-                for iterator in last_output_iterators:
-                    while(iterator.has_next()):
-                        on_atom_output(next(iterator))
+        for filter in self.__layers[len(self.__layers) - 1]:
```
Comment (CremaLuca) [inline otri/filtering/filter_list.py:76]: Codacy Issue found: [Redefining built-in 'filter'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/math_filter.py [hunk 15]
```diff
@@ -1,42 +1,60 @@
...
     Input:
-        Single stream 
+        Single stream.
     Output:
-        Single stream
+        Single stream.
     '''
 
-    def __init__(self, input_stream: Stream, keys_operations: Mapping[str, Callable]):
+    def __init__(self, input: str, output: str, keys_operations: Mapping[str, Callable]):
```
Comment (CremaLuca) [inline otri/filtering/filters/math_filter.py:14]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/split_filter.py [hunk 16]
```diff
@@ -16,38 +15,70 @@ class SplitFilter(Filter):
...
         )
         self.__key = key
         self.__side = side
         self.__ranges = ranges
-        self.__ignore_none = ignore_none
-        self.__source_iter = source_stream.__iter__()
-        if not ignore_none:
-            self.__none_stream = self.get_output_stream(n+1)
+
+    def setup(self, inputs : Sequence[Stream], outputs : Sequence[Stream], status: Mapping[str, Any]):
```
Comment (CremaLuca) [inline otri/filtering/filters/split_filter.py:62]: Codacy Issue found: [Signature differs from overridden 'setup' method](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/nuplicator_filter.py [hunk 17]
```diff
@@ -1,31 +1,49 @@
...
-    Outputs: Any number of streams.
+
+    Input: 
+        Single stream.
+    Outputs:
+        Any number of streams.
     '''
 
-    def __init__(self, source_stream: Stream, output_streams_count: int, deep_copy: bool = True):
+    def __init__(self, input: str, output: Sequence[str], deep_copy: bool = True):
```
Comment (CremaLuca) [inline otri/filtering/filters/nuplicator_filter.py:15]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/interpolation_filter.py [hunk 18]
```diff
@@ -14,56 +14,75 @@
...
+            output_count=1
+        )
         self.target_interval = target_interval
-        self.keys_to_change = keys_to_change
+        self.keys_to_interp = keys_to_interp
         self.timeunit = InterpolationFilter.__timedelta_from_interval(
             interval=target_interval)
         self.atom_buffer = None
 
+    def setup(self, inputs: Sequence[Stream], outputs: Sequence[Stream], status: Mapping[str, Any]):
```
Comment (CremaLuca) [inline otri/filtering/filters/interpolation_filter.py:49]: Codacy Issue found: [Signature differs from overridden 'setup' method](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/interpolation_filter.py [hunk 19]
```diff
@@ -14,56 +14,75 @@
...
-    The resulting atoms will have given values interpolated
+    The resulting atoms will have given values interpolated.
     Input:
         Oredered by datetime atoms.
     Output:
         Atoms interpolated for the given target interval
     '''
 
-    def __init__(self, input_stream: Stream, keys_to_change: Collection[str], target_interval: str = "minutes"):
+    def __init__(self, input: str, output: str, keys_to_interp: Collection[str], target_interval: str = "minutes"):
```
Comment (CremaLuca) [inline otri/filtering/filters/interpolation_filter.py:24]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/phase_filter.py [hunk 20]
```diff
@@ -69,46 +90,53 @@ class PhaseMulFilter(PhaseFilter):
     Input:
         Single stream ordered by datetime
     Output:
-        A single stream containing n - c atoms
+        Single stream containing n - c atoms
     '''
 
-    def __init__(self, input_stream: Stream, keys_to_change: Collection[str], distance: int):
+    def __init__(self, input: str, output: str, keys_to_change: Collection[str], distance: int):
```
Comment (CremaLuca) [inline otri/filtering/filters/phase_filter.py:96]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filters/merge_filter.py [hunk 21]
```diff
@@ -1,41 +1,59 @@
...
+    Sequentially merges elements from multiple streams into one single output.
+    
     Input:
         Multiple streams.
     Output:
         A single stream containing data read sequentially (all of stream 1, then all of stream 2 and so on).
     '''
 
-    def __init__(self, input_streams: Collection[Stream]):
+    def __init__(self, input: Sequence[str], output: str):
```
Comment (CremaLuca) [inline otri/filtering/filters/merge_filter.py:14]: Codacy Issue found: [Redefining built-in 'input'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 2
           

Complexity increasing per file
==============================
- test/filtering/filters/split_filter_test.py  1
- test/filtering/filter_list_test.py  1
         

Complexity decreasing per file
==============================
+ otri/filtering/filters/statistics_filter.py  -1
+ autocorrelation.py  -2
+ test/filtering/filter_test.py  -1
+ otri/filtering/filter_list.py  -7
         

Clones added
============
- otri/filtering/filters/statistics_filter.py  1
- otri/filtering/filters/generic_filter.py  1
- otri/filtering/filters/nuplicator_filter.py  2
- otri/filtering/filters/phase_filter.py  2
- test/filtering/filters/split_filter_test.py  2
- otri/filtering/filters/interpolation_filter.py  2
- test/filtering/filters/merge_filter_test.py  3
- otri/filtering/filters/math_filter.py  2
         

Clones removed
==============
+ test/filtering/filters/phase_filter_test.py  -3
+ test/filtering/filters/statistics_filter_test.py  -2
+ autocorrelation.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486&bid=18499171)

File: test/filtering/filter_list_test.py [hunk 22]
```diff
@@ -0,0 +1,17 @@
+from otri.filtering.filter_list import Stream, FilterList, FilterLayer
```
Comment (CremaLuca) [inline test/filtering/filter_list_test.py:1]: Codacy Issue found: [Unused FilterLayer imported from otri.filtering.filter_list](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

File: otri/filtering/filter_list.py [hunk 23]
```diff
@@ -1,90 +1,92 @@
...
```
Comment (CremaLuca) [inline otri/filtering/filter_list.py:76]: Codacy Issue found: [Redefining built-in 'filter'](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
- Added 1
           

Complexity increasing per file
==============================
- test/filtering/filter_list_test.py  1
- test/filtering/filters/split_filter_test.py  1
         

Complexity decreasing per file
==============================
+ test/filtering/filter_test.py  -1
+ otri/filtering/filters/statistics_filter.py  -1
+ autocorrelation.py  -2
+ otri/filtering/filter_list.py  -7
         

Clones added
============
- otri/filtering/filters/nuplicator_filter.py  2
- test/filtering/filters/merge_filter_test.py  3
- otri/filtering/filters/math_filter.py  2
- otri/filtering/filters/phase_filter.py  2
- test/filtering/filters/split_filter_test.py  2
- otri/filtering/filters/interpolation_filter.py  2
- otri/filtering/filters/generic_filter.py  1
- otri/filtering/filters/statistics_filter.py  1
         

Clones removed
==============
+ test/filtering/filters/phase_filter_test.py  -3
+ autocorrelation.py  -2
+ test/filtering/filters/statistics_filter_test.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486&bid=18499171)

File: test/filtering/filter_list_test.py (hunk 22 above)
Comment (CremaLuca) [inline test/filtering/filter_list_test.py:1]: Codacy Issue found: [Unused FilterLayer imported from otri.filtering.filter_list](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- test/filtering/filter_list_test.py  1
- test/filtering/filters/split_filter_test.py  1
         

Complexity decreasing per file
==============================
+ autocorrelation.py  -2
+ otri/filtering/filters/statistics_filter.py  -1
+ test/filtering/filter_test.py  -1
+ otri/filtering/filter_list.py  -7
         

Clones added
============
- otri/filtering/filters/phase_filter.py  2
- otri/filtering/filters/math_filter.py  2
- otri/filtering/filters/interpolation_filter.py  2
- otri/filtering/filters/statistics_filter.py  1
- test/filtering/filters/merge_filter_test.py  3
- otri/filtering/filters/generic_filter.py  1
- test/filtering/filters/split_filter_test.py  2
- otri/filtering/filters/nuplicator_filter.py  2
         

Clones removed
==============
+ autocorrelation.py  -2
+ test/filtering/filters/statistics_filter_test.py  -2
+ test/filtering/filters/phase_filter_test.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486&bid=18499171)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- test/filtering/filters/split_filter_test.py  1
- test/filtering/filter_list_test.py  1
         

Complexity decreasing per file
==============================
+ otri/filtering/filter_list.py  -7
+ autocorrelation.py  -2
+ test/filtering/filter_test.py  -1
+ otri/filtering/filters/statistics_filter.py  -1
         

Clones added
============
- otri/filtering/filters/math_filter.py  2
- test/filtering/filters/merge_filter_test.py  3
- test/filtering/filters/split_filter_test.py  2
- otri/filtering/filters/nuplicator_filter.py  2
- otri/filtering/filters/generic_filter.py  1
- otri/filtering/filters/phase_filter.py  2
- otri/filtering/filters/interpolation_filter.py  2
- otri/filtering/filters/statistics_filter.py  1
         

Clones removed
==============
+ test/filtering/filters/statistics_filter_test.py  -2
+ test/filtering/filters/phase_filter_test.py  -3
+ autocorrelation.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486&bid=18499171)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- test/filtering/filters/split_filter_test.py  1
- test/filtering/filter_list_test.py  1
         

Complexity decreasing per file
==============================
+ test/filtering/filter_test.py  -1
+ otri/filtering/filter_list.py  -7
+ otri/filtering/filters/statistics_filter.py  -1
+ autocorrelation.py  -2
         

Clones added
============
- test/filtering/filters/merge_filter_test.py  3
- otri/filtering/filters/phase_filter.py  2
- test/filtering/filters/split_filter_test.py  2
- otri/filtering/filters/generic_filter.py  1
- otri/filtering/filters/statistics_filter.py  1
- otri/filtering/filters/nuplicator_filter.py  2
- otri/filtering/filters/math_filter.py  2
- otri/filtering/filters/interpolation_filter.py  2
         

Clones removed
==============
+ test/filtering/filters/phase_filter_test.py  -3
+ test/filtering/filters/statistics_filter_test.py  -2
+ autocorrelation.py  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486&bid=18499171)

**Comment by CremaLuca** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 6
           

Complexity increasing per file
==============================
- test/filtering/filters/split_filter_test.py  1
- test/filtering/filter_list_test.py  1
- otri/filtering/filter_net.py  6
         

Complexity decreasing per file
==============================
+ test/filtering/filter_test.py  -1
+ autocorrelation.py  -2
+ otri/filtering/filters/statistics_filter.py  -1
         

Clones added
============
- test/filtering/filters/merge_filter_test.py  3
- otri/filtering/filters/phase_filter.py  2
- test/filtering/filters/split_filter_test.py  2
- otri/filtering/filters/generic_filter.py  1
- otri/filtering/filters/statistics_filter.py  1
- otri/filtering/filters/interpolation_filter.py  2
- otri/filtering/filters/math_filter.py  2
- otri/filtering/filters/nuplicator_filter.py  2
         

Clones removed
==============
+ autocorrelation.py  -2
+ test/filtering/filters/statistics_filter_test.py  -2
+ test/filtering/filters/phase_filter_test.py  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/OTRI-Unipd/OTRI/pullRequest?prid=5649486&bid=18499171)

File: otri/downloader/gme_downloader.py [hunk 24]
```diff
@@ -171,7 +171,7 @@ def __get_data(category: str, req_type: str, day: date) -> Union[str, bool]:
         url = "https://www.mercatoelettrico.org/It/Tools/Accessodati.aspx?ReturnUrl=/It/WebServerDataStore/{}_{}/{}{}{}.xml".format(
             category, req_type, formatted_date, category, req_type)
         response = session.post(url, data=post_data)
-        if(response.status_code == 200):
+        if(response.state_code == 200):
```
Comment (ricdezen) [inline otri/downloader/gme_downloader.py]: Is this alright?

**Review by ricdezen — COMMENTED** [review]
I thought another 1k lines got added, it's just the log file. Heavily suggest adding that to the .gitignore.
Aside from something on gmedownloader that I pointed out I have no complaints, I was using this branch when refactoring the validator and everything seemed ok

**Comment by CremaLuca** [comment]
> Heavily suggested adding (log.txt) to the .gitignore

BOI
image

Update: Looks like log files were left when changing branch but weren't handled by gitignore

File: otri/downloader/gme_downloader.py (hunk 24 above)
Comment (CremaLuca) [reply otri/downloader/gme_downloader.py]: Whoops auto-renaming

**Review by ricdezen — APPROVED** [review]
Nice

**Pull request merged**